materials/variation_with_state.py

```diff
@@ -0,0 +1,254 @@
+    """Create 2d interpolation arrays for `griddata` from a YAML dict.
+
+    Arguments:
+        yaml_dict (dict): variation with state subdictionary extracted from a YAML file.
+        state_vars (list of string): State variable names.
+        state_vars_interp_scales (list of string): Interpolation scale for each state
+            variable. Should be 'log' or 'linear'.
+
+    Returns:
+        ndarray: interpolation points.
+        ndarray: interpolation values.
+    """
+    if len(state_vars) != 2:
+        raise ValueError()
+    # TODO there has got to be a less awful way to represent a 2+ dimensional
+    # interpolation table in YAML and load it.
+    interp_points = np.empty((0, 2))
+    interp_values = np.array([])
+    tbl_dict = yaml_dict[state_vars[0]]
+    interp_points_0 = np.array(list(tbl_dict.keys()))
+    interp_points_0.sort()
+    interp_points_1 = np.array(
+        [tbl_dict[p0][state_vars[1]] for p0 in interp_points_0])
+    # From the yaml dictionary 2d table, build an array compatible with
+    # scipy.interpolate.griddata
+    for i in range(len(interp_points_0)):
+        # The value of the 0th state variable for this slice of the interpolation table.
+        point_0 = interp_points_0[i]
+        interp_points = np.append(
+            interp_points,
+            np.stack(
+                (
+                    np.full_like(interp_points_1[i], point_0, dtype=np.double),
+                    interp_points_1[i]
+                ),
+                axis=1),
+            axis=0)
+        interp_values = np.append(interp_values, tbl_dict[point_0]['values'])
+    for j in range(2):
+        if state_vars_interp_scales[j] == 'log':
+            interp_points[:, j] = np.log(interp_points[:, j])
+
+    return interp_points, interp_values
+
+
+def _create_interp_arrays_from_yaml_table(yaml_dict, state_vars, state_vars_interp_scales):
+    """Create 1d or 2d interpolation arrays for `griddata` from a YAML dict.
+
+    Arguments:
+        yaml_dict (dict): property dictionary extracted from a YAML file.
+        state_vars (list of string): State variable names.
+        state_vars_interp_scales (list of string): Interpolation scale for each state
+            variable. Should be 'log' or 'linear'.
+    Returns:
+        ndarray: interpolation points.
+        ndarray: interpolation values.
+    """
+    if len(state_vars) == 1:
+        interp_points = np.array(
+            yaml_dict[state_vars[0]])
+        interp_values = np.array(
+            yaml_dict['values'])
+        if state_vars_interp_scales[0] == 'log':
+            interp_points = np.log(interp_points)
+    elif len(state_vars) == 2:
+        interp_points, interp_values = _create_interp_arrays_from_yaml_table_2d(
+            yaml_dict, state_vars, state_vars_interp_scales)
+    else:
+        raise NotImplementedError('More than two state variables not supported.')
+
+    return interp_points, interp_values
+
+
+class VariationWithState:
```

Comment:
Prefer to inherit from `object` (Unless you have an unusual reason not to!)

Reply:
I didn't think I needed to do so in python 3?
https://stackoverflow.com/questions/1238606/is-it-necessary-or-useful-to-inherit-from-pythons-object-in-python-3-x
Do you think it's still good to do so in python 3?

Reply:
Oops, you're right!